_posts/2021-11-02-introducing-the-design-for-stargate-v2.md 

```diff
@@ -0,0 +1,182 @@
+
+
+### Load balancing, authentication, and other cross-cutting concerns
+
+We don’t assume usage of any particular ingress, but encourage its use. The intent is to be compatible with whatever load balancer is used, be it Nginx, Envoy, or HAProxy.
+
+For HTTP services we could move some of the cross-cutting functionality into a load balancer by exposing endpoints for rate limiting and ext_authz. But we would then leave it up to each user to implement this for their particular load balancer (assuming that their load balancer supports that functionality). For CQL, we still need to implement this logic within Stargate since load balancers wouldn’t be able to act on the binary protocol.
+
+Authentication and authorization will continue to be pluggable as it is today. Although we include some base implementations, it’ll still be possible to support other methods by implementing the necessary interface.
+
+
+## Deployment considerations
+
+Breaking apart Stargate will require changes to how it’s deployed, meaning we’ll need to create a Helm chart. As we iterate, this chart can become more advanced and allow for a more customizable deployment (e.g. just REST or some other subset of services).
+
+In the base case, the chart will deploy each of the user-facing services as a separate pod in addition to the Coordinator Node pod (persistence, CQL, and Bridge). For resource constrained environments, where there’s a sensitivity to the number of pods, various combinations of pods and containers are possible. For one, it could maintain the same setup as exists today by placing all of the containers in a single pod. Another setup could be two pods: one for the Coordinator Node and another for the services.
+
+
+## Final notes
+
+We believe that the changes proposed in this design will lead to a much better experience for those who contribute to the Stargate project and those who run it to support their other applications. 
+
+Contributors will be able to quickly iterate on just the parts of the project they want — without needing to understand as deeply the other components. Plus, it’ll open up the potential for newer Java versions and frameworks rather than Java 8 with OSGi or even a polyglot environment. Meanwhile, operators will have greater flexibility over their Stargate deployment and the potential for a lower-resource footprint when scaling to handle higher loads. 
+
+Overall, we’re excited about these new changes and look forward to working with the [Stargate community](https://dtsx.io/3pMAp0Y) to make this new design a reality. On that note, if you want to review this design and give us your feedback, [join the discussion on GitHub](https://github.com/stargate/stargate/discussions/1381). \
```

Comment:
Remove trailing backslash on this line?

Reply:
Missed that one. Not sure why the docs -> markdown randomly adds those.